### PR TITLE
Make loading a bit more lazy and side effecting things more conservative (buffer local)

### PR DESCRIPTION
**Pull request by bjeanes**
I spot checked most features that I use and everything still seems to work but let me know if there are issues with these changes.

### CODE REVIEW

File: ftplugin/crystal.vim [hunk 1]
```diff
@@ -59,7 +59,7 @@ nnoremap <Plug>(crystal-format) :<C-u>CrystalFormat<CR>
 
 augroup plugin-ft-crystal
     autocmd!
-    autocmd BufWritePre * if &ft ==# 'crystal' && g:crystal_auto_format | CrystalFormat | endif
+    autocmd BufWritePre <buffer> if &ft ==# 'crystal' && g:crystal_auto_format | CrystalFormat | endif
```
Comment (rhysd) [inline ftplugin/crystal.vim]: If making this auto command buffer-local, checking `&ft` is redundant, I think.  Although it is edge case where user changes its filetype after loading the file as crystal code.

Comment (bjeanes) [inline ftplugin/crystal.vim]: I had the same thinking. I opted to leave it to avoid any surprises, but I agree that it's redundant 95%+ of the time.

Comment (rhysd) [inline ftplugin/crystal.vim]: OK, could you please remove `&ft` check?

Comment (bjeanes) [inline ftplugin/crystal.vim]: Done!

**Comment by rhysd** [comment]
@bjeanes Thanks!

**Pull request merged**